### ISSUE TITLE
Question: Why doesn't Carlo automatically install Chrome in the home directory?

### ISSUE THREAD

**Issue by kitze** [GoogleChromeLabs/carlo#90]
Hey.

So this is a really great initiative and I'm following the progress closely. I was wondering if it would be a good idea for carlo apps to ship with a small installer which will check if Chrome is available, and if it's not, it's gonna download a version and install it in the home directory. The user doesn't have to be aware of this logic, just like he's not aware that Electron apps are shipping with a Chromium instance.

What do you think?

**Comment by pavelfeldman** [comment]
Chrome should only be installed with the user consent. We can print error message or point user to the installation, but user needs to be aware of what is happening and why.

**Comment by ab35** [comment]
It would be great to have chromium embedded by default in all the operating systems, as a platform standard, with an option to upgrade to chrome if the user wants to. Ha ha.

**Comment by pavelfeldman** [comment]
Unlike Chrome, Chromium is just an open source library that one can use, no end user agreement is necessary.

**Comment by kitze** [comment]
@pavelfeldman yeah the end goal is for Carlo to work when the user doesn't have Chrome/Chromium. So is an automatic download of Chromium supported now?

**Comment by ab35** [comment]
@kitze If one chooses a chromium channel in the [launch options](https://github.com/GoogleChromeLabs/carlo/blob/master/API.md#carlolaunchoptions), then carlo calls puppeteer [createBrowserFetcher](https://github.com/GoogleChromeLabs/carlo/blob/7b0d2821694caef7c24cd3fc960cb0657b377780/lib/find_chrome.js#L198) which will download the chromium revision compatible with Puppeteer or even a specific revision.

That is cool, but chromium has a disk and bandwidth cost:  (~170MB Mac, ~282MB Linux, ~280MB Win) and apparently one can only update the whole chromium package because updates do not come by little patches on top of the existing installation.

It would be cool if the current installation was able to update just the least amount of bytes by fetching only the required diff instead of having to download the whole package each time.

**Comment by ab35** [comment]
@pavelfeldman  is chromium [auto-update feature](https://www.chromium.org/chromium-os/chromiumos-design-docs/autoupdate-details) using "_filesystem updates with file-level binary diffs_" available in carlo / puppeteer-core ?

**Comment by ChrisKader** [comment]
Chromium is bundled with the application and is only used by the applicatiion. It’s part of the application that makes it function. That is also way electron apps are very large. 

Where as with Carlo, you don’t need to download 100MB+ apps every time, eve though they all use the same runtime. Carlo let’s you distribute much smaller apps that use one Chrome install.

I view Carlo as something very similar to ADOBE AIR or FLASH, to a point. Use a common, preinstalled runtime, to distribute much smaller applications.

**Comment by trusktr** [comment]
@ab35 Just to verify, you're saying that the user need not have Chrome or Chromium installed, and that with the launch option you recommended it will automatically install Chromium behind the scenes, without the user knowing, and launch the Carlo app?

Will this automatically-installed Chromium instance be the accessible as an app on the User's desktop? I don't want that to be the case, just want the user to install "one app" not "two apps".

Do you have an example?

**Comment by ab35** [comment]
@trusktr If I correctly remember, when the user does not have Chrome or Chromium, one can programmatically download a specific version (see one's comment above), and that version is downloaded somewhere into the carlo application directory tree. It will not appear on the User's desktop or as an installed app, because it just downloaded into a local folder.
